test/core/test_directional_contact_interaction.cpp

```diff
@@ -0,0 +1,230 @@
+#define BOOST_TEST_MODULE "test_directional_contact_interaction"
+
+#ifdef BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
+#else
+#include <boost/test/included/unit_test.hpp>
+#endif
+
+#include <mjolnir/core/BoundaryCondition.hpp>
+#include <mjolnir/core/SimulatorTraits.hpp>
+#include <mjolnir/interaction/local/DirectionalContactInteraction.hpp>
+#include <mjolnir/potential/local/CosinePotential.hpp>
+#include <mjolnir/potential/local/GoContactPotential.hpp>
+#include <mjolnir/util/make_unique.hpp>
+#include <mjolnir/math/constants.hpp>
+#include <mjolnir/math/Matrix.hpp>
+
+#include <tuple>
+#include <random>
+
+BOOST_AUTO_TEST_CASE(DirectionalContactInteraction_numerical_diff)
+{
+  mjolnir::LoggerManager::set_default_logger("test_directional_contact_interaction.log");
+  using traits_type              = mjolnir::SimulatorTraits<double, mjolnir::UnlimitedBoundary>;
+  using real_type                = traits_type::real_type;
+  using coord_type               = traits_type::coordinate_type;
+  using boundary_type            = traits_type::boundary_type;
+  using system_type              = mjolnir::System<traits_type>;
+  using angle_potential_type     = mjolnir::CosinePotential<real_type>;
+  using contact_potential_type   = mjolnir::GoContactPotential<real_type>;
+  using directional_contact_type = mjolnir::DirectionalContactInteraction<
+    traits_type, angle_potential_type, angle_potential_type, contact_potential_type>;
+
+  const real_type pi = mjolnir::math::constants<real_type>::pi();
+  const real_type k_angle(1e0);
+  const real_type k_contact(1e0);
+  const real_type angle1_native(pi);
+  const real_type angle2_native(pi);
+  const real_type contact_native(2.0);
+  angle_potential_type angle1_potential    = angle_potential_type{k_angle, 1, angle1_native};
```

Comment:
Is there any reason to keep it non-const?
It is a test, so you need to ensure that accidental changes will not occur after here.
If you don't have any reason, add const.
Also, you don't need to write `angle_potential_type` twice.
```suggestion
const auto angle1_potential = angle_potential_type{k_angle, 1, angle1_native};
```